designs/karpenter-integration-testing.md

```diff
@@ -0,0 +1,271 @@
+# Karpenter - Integration Testing RFC
+
+Note: This document serves as a living specification for Karpenter's integration testing strategy and orchestration mechanisms. As the project evolves, both the design requirements and this documentation will be updated to reflect current best practices and project needs.
```

Comment:
Will this document be expanded in the future to detail how a cloudprovider can use the upstream integration tests in their own environment with their own NodeClass?

Reply:
No, we will not be including that process in this document. We will have a living README document in the test suite that will outline how cloud provider and utilize these tests